src/main/java/uk/ac/ebi/eva/pipeline/io/processors/VariantStatsProcessor.java

```diff
@@ -0,0 +1,186 @@
+package uk.ac.ebi.eva.pipeline.io.processors;
+
+import com.mongodb.BasicDBObject;
+import org.opencb.biodata.models.feature.Genotype;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.batch.item.ItemProcessor;
+import uk.ac.ebi.eva.commons.models.data.VariantStats;
+import uk.ac.ebi.eva.commons.models.mongo.entity.VariantDocument;
+import uk.ac.ebi.eva.commons.models.mongo.entity.subdocuments.VariantSourceEntryMongo;
+import uk.ac.ebi.eva.commons.models.mongo.entity.subdocuments.VariantStatsMongo;
+import uk.ac.ebi.eva.pipeline.io.readers.VariantStatsReader;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class VariantStatsProcessor implements ItemProcessor<VariantDocument, VariantDocument> {
```

Comment:
In case any of the variant does not have the field "files" or if the field "files" is empty, the variant will not be processed and the variant will be returned as it is. 
Same will happen in case any object in the files array does not have sample data, calculation of stats for that object will be skipped.